web/src/connectors/documents-page-control-connector/use-notify-processing.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react';
+import { NotificationTypes, showNotification } from 'shared/components/notifications';
+
+const notifyProcessing = () => {
+    showNotification({
+        type: NotificationTypes.hint,
+        text: 'Your file is still processing. This may take a while'
+    });
+};
+
+export const useNotifyProcessing = (isUploaded: boolean) => {
+    useEffect(() => {
+        let delay = 10000;
+        let timeoutId: number | null = null;
+        let firstNotificationShown = false;
+
+        const showNextNotification = () => {
+            if (isUploaded) {
+                if (!firstNotificationShown) {
+                    timeoutId = setTimeout(() => {
+                        firstNotificationShown = true;
+                        showNextNotification();
+                    }, delay) as unknown as number;
```

Comment:
Instead of doing this casting, I propose to use "ReturnType<typeof setTimeout> | null" instead of "number | null" for "timeoutId"